src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -739,7 +739,6 @@ public async Task<ILaunchSettings> WaitForFirstSnapshot(int timeout)
 
             await _firstSnapshotCompletionSource.Task.TryWaitForCompleteOrTimeout(timeout);
 
-            Assumes.NotNull(CurrentSnapshot);
```

Comment:
Would it make sense to enforce this if TryWaitForCompleteOrTimeout returns true? The annotation in various callers seem to have this expectation.

Reply:
That's reasonable/